src/analysis/normed_space/add_torsor.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2020 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Joseph Myers.
+-/
+import algebra.add_torsor
+import analysis.normed_space.basic
+
+noncomputable theory
+
+/-!
+# Torsors of additive normed group actions.
+
+This file defines torsors of additive normed group actions, with a
+metric space structure.  The motivating case is Euclidean affine
+spaces.
+
+-/
+
+universes u v
+
+section prio
+set_option default_priority 100 -- see Note [default priority]
+/-- A `normed_add_torsor V P` is a torsor of an additive normed group
+action by a `normed_group V` on points `P`. We bundle the metric space
+structure and require the distance to be the same as results from the
+norm (which in fact implies the distance yields a metric space, but
+bundling just the distance and using an instance for the metric space
+results in type class problems). -/
+class normed_add_torsor (V : Type u) (P : Type v) [normed_group V] [metric_space P]
+  extends add_torsor V P :=
+(torsor_dist_eq_norm' : ∀ (x y : P), dist x y = ∥(x -ᵥ y : V)∥)
+end prio
+
+/-- The distance equals the norm of subtracting two points. This lemma
+is needed to make V an explicit rather than implicit argument. -/
+lemma torsor_dist_eq_norm (V : Type u) {P : Type v} [normed_group V] [metric_space P]
```

Comment:
```suggestion
lemma dist_eq_norm (V : Type u) {P : Type v} [normed_group V] [metric_space P]
```

Reply:
Maybe this gives a name conflict... in that case I suggest appending a prime `'`, or putting it in a namespace: `add_torsor.dist_eq_norm`.

Reply:
I've put it in a namespace.